### PR TITLE
feat: add prune for ancient sidecars data

### PR DESCRIPTION
**Pull request by pythonberg1997**
### Description

This pr is to add prune for ancient sidecars data

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add prune for ancient sidecars data

### Potential Impacts
* add potential impacts for other components here
* ...

### CODE REVIEW

File: crates/config/src/config.rs [hunk 1]
```diff
@@ -332,14 +332,16 @@ impl Default for IndexHistoryConfig {
 pub struct PruneConfig {
     /// Minimum pruning interval measured in blocks.
     pub block_interval: usize,
+    /// The number of ancient sidecars to keep in the static file provider.
+    pub ancient_sidecars_kept_blocks: usize,
```
Comment (unclezoro) [inline crates/config/src/config.rs]: can we change `ancient_sidecars_kept_blocks` to `recent_sidecars_kept_blocks`, I think it is easier to understand.

File: crates/config/src/config.rs [hunk 2]
```diff
@@ -332,14 +332,16 @@ impl Default for IndexHistoryConfig {
...
     /// Pruning configuration for every part of the data that can be pruned.
     #[serde(alias = "parts")]
     pub segments: PruneModes,
 }
 
 impl Default for PruneConfig {
     fn default() -> Self {
-        Self { block_interval: 5, segments: PruneModes::none() }
+        Self { block_interval: 5, ancient_sidecars_kept_blocks: 0, segments: PruneModes::none() }
```
Comment (unclezoro) [inline crates/config/src/config.rs]: what does "0" means,  do not keep or keep all sidecars?

File: crates/prune/prune/src/pruner.rs [hunk 3]
```diff
@@ -328,6 +345,76 @@ impl<DB: Database> Pruner<DB> {
...
+        );
+
+        while range_start > 0 {
+            let range = find_fixed_range(range_start - 1);
+            let path =
+                static_file_provider.path().join(StaticFileSegment::Sidecars.filename(&range));
+
+            if path.exists() {
+                delete_static_files(&path);
+                self.metrics.ancient_sidecars_file.increment(1);
```
Comment (unclezoro) [inline crates/prune/prune/src/pruner.rs]: `ancient_sidecars_file` seems a bit useless, what about a metrics about the oldest sidecar height?

File: crates/config/src/config.rs (hunk 1 above)
Comment (pythonberg1997) [reply crates/config/src/config.rs]: Done

File: crates/config/src/config.rs (hunk 2 above)
Comment (pythonberg1997) [reply crates/config/src/config.rs]: keep all sidecars

File: crates/prune/prune/src/pruner.rs (hunk 3 above)
Comment (pythonberg1997) [reply crates/prune/prune/src/pruner.rs]: Done

**Review by unclezoro — APPROVED** [review]

**Pull request merged**